### ISSUE TITLE
Linting on Jupyter Notebook fails after reordering cells

### ISSUE THREAD

**Issue by michue-work** [astral-sh/ruff-vscode#566]
Hello, today I activated Ruff for Jupyter Notebooks and noticed some strange behavior I didn't see with normal Python files:

_After reordering the cells of a notebook, the editor window shows false-positives from Ruff linting._

**MWE**
`pyproject.toml`
```
[tool.ruff]
extend-include = [
    "*.ipynb",
]
```

`ruff.ipynb` (with --- denoting a new cell)
```
import random
---
f = random.random()
---
g = random.random()
```

After moving the third cell with `g = ...` to the second position, Ruff reports F401 on `import random`.

**Environment**
Operating System Windows_NT x64 10.0.19045
VS Code 1.91.1 (commit f1e16e1e6214d7c44d078b1f0607b2388f29d729)
Ruff extension v2024.36.0

**Comment by dhruvmanila** [comment]
Hey, thanks for the issue. Can you expand on how you're reordering the cells? I'm unable to do so just by dragging it around:

https://github.com/user-attachments/assets/cda10601-2428-449e-8f2f-b257d8d01679

**Comment by michue-work** [comment]
Hey Dhruv,  
I did basically the same as you. You can see it in the screen recording:

[Video 7-29 at 18.00.webm](https://github.com/user-attachments/assets/0108846a-2bc7-4b05-b95d-b38fd5c6e0e9)

**Comment by michue-work** [comment]
I also deactivated all VS Code extensions except for Python, Python Debugger and Ruff itself - still the same behavior.

But as a remark, the behavior is a little elusive - it does not show up consistently. And usually closing and reopening the file resolves the issue, but sometimes.  
I am sorry, that can't pin it down more precisely ...

**Comment by dhruvmanila** [comment]
Thanks for providing the video.

> I also deactivated all VS Code extensions except for Python, Python Debugger and Ruff itself - still the same behavior.

I can try doing this and see if that reproduces the issue.

**Comment by dhruvmanila** [comment]
Huh, I can reproduce it now.

**Comment by dhruvmanila** [comment]
Never mind, I can reproduce it even if all the extensions are enabled.

**Comment by dhruvmanila** [comment]
I suspect that there's some logic error in the `notebookDocument/didChange` request handling.

**Comment by michue-work** [comment]
I'm quite happy, that you were able to reproduce this behavior!  
If I can help in narrowing the problem further down, please let me know.
